File: labs/solidworks/swex/pmpage/events/closing/Events.Closed.cs
public class DataModel
{
    public string Text { get; set; }
}

private DataModel m_Data;

private PropertyManagerPageEx<MyPMPageHandler, DataModel> m_Page;

public override bool OnConnect()
{
    m_Data = new DataModel();
    m_Page = new PropertyManagerPageEx<MyPMPageHandler, DataModel>(App);

    m_Page.Handler.Closed += OnClosed;

    return true;
}

private void OnClosed(swPropertyManagerPageCloseReasons_e reason)
{
    if (reason == swPropertyManagerPageCloseReasons_e.swPropertyManagerPageClose_Okay)
    {
        //TODO: do work
    }
    else
    {
        //TODO: release resources
    }
}

File: labs/solidworks/swex/pmpage/events/closing/Events.Closed.vb
Public Class DataModel
    Public Property Text As String
End Class

Private m_Data As DataModel
Private m_Page As PropertyManagerPageEx(Of MyPMPageHandler, DataModel)

Public Overrides Function OnConnect() As Boolean
    m_Data = New DataModel
    m_Page = New PropertyManagerPageEx(Of MyPMPageHandler, DataModel)(App)

    AddHandler m_Page.Handler.Closed, AddressOf OnClosed
    Return True
End Function

Private Sub OnClosed(ByVal reason As swPropertyManagerPageCloseReasons_e)
    If reason = swPropertyManagerPageCloseReasons_e.swPropertyManagerPageClose_Okay Then
        'TODO: do work
    Else
        'TODO: release resources
    End If
End Sub